src/disco/reducers/telemetry.js

```diff
@@ -0,0 +1,46 @@
+/* @flow */
+import invariant from 'invariant';
+
+export const SET_HASHED_CLIENT_ID: 'SET_HASHED_CLIENT_ID' =
+  'SET_HASHED_CLIENT_ID';
+
+export type TelemetryState = {|
+  hashedClientId?: string,
+|};
+
+export const initialState: TelemetryState = {
+  hashedClientId: undefined,
+};
+
+type SetHashedClientIdAction = {|
+  type: typeof SET_HASHED_CLIENT_ID,
+  payload: { hashedClientId: string },
+|};
+
+export const setHashedClientId = (
+  hashedClientId: string,
+): SetHashedClientIdAction => {
+  if (!hashedClientId) {
+    invariant(hashedClientId, 'hashedClientId is required');
```

Comment:
Why do we need the if here?

Reply:
ok I can remove this. I just thought it was an extra check but maybe it's not necessary..

Reply:
invariant should be enough

Reply:
ahhh I see:)